src/sentry/api/endpoints/organization_transaction_details.py

```diff
@@ -0,0 +1,314 @@
+from collections import defaultdict
+from datetime import datetime, timedelta, timezone
+
+from rest_framework.request import Request
+from rest_framework.response import Response
+from sentry_relay.consts import SPAN_STATUS_CODE_TO_NAME
+from snuba_sdk import Column
+
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases.organization import OrganizationEndpoint
+from sentry.constants import ObjectStatus
+from sentry.models.project import Project
+from sentry.search.events.builder.spans_indexed import SpansIndexedQueryBuilder
+from sentry.search.events.types import ParamsType
+from sentry.snuba.dataset import Dataset
+from sentry.snuba.referrer import Referrer
+
+
+@region_silo_endpoint
+class OrganizationTransactionDetailsEndpoint(OrganizationEndpoint):
+    owner = ApiOwner.PERFORMANCE
+    publish_status = {
+        "GET": ApiPublishStatus.PRIVATE,
+    }
+
+    def get(self, request: Request, organization, project_slug, transaction_id):
+        try:
+            project = Project.objects.get(
+                slug=project_slug, organization_id=organization.id, status=ObjectStatus.ACTIVE
+            )
+        except Project.DoesNotExist:
+            return Response(status=404)
+
+        # Check access to the project as this endpoint doesn't use membership checks done
+        # get_filter_params().
+        if not request.access.has_project_access(project):
+            return Response(status=404)
+
+        # Get a list of all spans with this transaction_id
+        start = request.GET.get("start_timestamp")
+        end = request.GET.get("end_timestamp")
+
+        spans_data = _query_all_spans_in_transaction(
+            organization, project, transaction_id, start, end
+        )
+        if len(spans_data) == 0:
+            return Response(status=404)
+
+        segment_span = next((span for span in spans_data if span["is_segment"] != 0), None)
+        if segment_span is None:
+            return Response({"detail": "No segment span found"}, status=500)
```

Comment:
what's a segment span?

Reply:
The short answer is, segment spans are how we're modelling Sentry's transaction concept after moving to a spans-only ingestion model.

The following spans are segment spans:
* root spans (spans with no parent)
* spans whose parent is in a different service

Segment spans will hold most of the data that we previously attached to transactions. The ID of the segment span will also be propagated to its children by our SDKs (which is why this endpoint is able to query spans by `segment_id`).

Even though the UI will continue to use the "transaction" name (which is why I'm using it for the endpoint name, parameter name, etc), really in the backend it's now a segment. (It is not clear to me why we chose to change the name - there's an idea floating around that maybe not all segments are transactions, but in practice I don't think that's currently true.)

Reply:
🤔 We should still make the error say `Transaction` and not `Segment` since we're not going to be communicating to the user about `segments`